src/descriptor/mod.rs

```diff
@@ -732,6 +732,34 @@ impl<Pk: MiniscriptKey> Descriptor<Pk> {
             )),
         }
     }
+
+    /// Whether the descriptor is safe
+    /// Checks whether all the spend paths in the descriptor are possible
+    /// on the bitcoin network under the current standardness and consensus rules
+    /// Also checks whether the descriptor requires signauture on all spend paths
+    /// And whether the script is malleable.
+    /// In general, all the guarantees of miniscript hold only for safe scripts.
+    /// All the analysis gurantees of miniscript only hold safe scripts.
+    /// The signer may not be able to find satisfactions even if one exists
+    pub fn sanity_check(&self) -> Result<(), Error> {
+        match *self {
+            Descriptor::Bare(ref ms) => ms.sanity_check()?,
+            Descriptor::Pk(ref _pk)
+            | Descriptor::Pkh(ref _pk)
+            | Descriptor::Wpkh(ref _pk)
+            | Descriptor::ShWpkh(ref _pk) => {}
+            Descriptor::Wsh(ref ms) | Descriptor::ShWsh(ref ms) => ms.sanity_check()?,
+            Descriptor::Sh(ref ms) => ms.sanity_check()?,
+            Descriptor::WshSortedMulti(ref svm) | Descriptor::ShWshSortedMulti(ref svm) => {
+                // extra allocation using clone allows us to reuse
+                // check safety function from Miniscript fragment instead
+                // of implemneting more code for safety checks
+                svm.clone().multi_ms().sanity_check()?
```

Comment:
I think you should pull this clone-and-sanity-check logic into its own method on `SortedMultiVec`. Even though it's internal I think it's confusing to have a method `multi_ms` which does not sort the keys (and which therefore does not match the actual conversion of the SVM to a miniscript)

Reply:
I tried doing this. The odd part about that I don't have access to `Ctx` to create a new miniscript which I do not have inside the SortedMultiVec

Reply:
You can parameterize the sanity-check function over `Ctx`, which will be a bit awkward to call but should work